rcongui/src/components/RconSettings/blacklist.js

```diff
@@ -68,16 +64,22 @@ class Blacklist extends React.Component {
               onChange={(e) => this.setState({ name: e.target.value })}
             />
         </Grid>
-        <Grid item xs={2} spacing={1} className={`${classes.padding} ${classes.margin}`} justify="center" alignContent="center">
             <Tooltip fullWidth title="Blacklisted players will instantly be banned when entering the server." arrow>
-                <Button 
-                        color="secondary"
-                        variant="outlined"
-                        disabled={steam_id == "" || reason == ""} onClick={this.blacklistPlayer}>
-                    Blacklist
-                </Button>
-            </Tooltip>
-        </Grid>
+                <Grid item xs={2} spacing={1} className={`${classes.padding} ${classes.margin}`} justify="center" alignContent="center">
+                    <Button 
+                       color="secondary"
+                       variant="outlined"
+                       disabled={!steam_id || !reason }
+                       onClick={
+                         () => { 
+                           submitBlacklistPlayer(steam_id, name, reason)
+                             .then((res) => !res.failed && this.setState({ steam_id: "", name: "", reason: "" }))
```

Comment:
Hmm, it may be a bit risky here to assume that this function with return a promise, or that it will return the results (in my experimental branch with the accounts, they won't)

Since the caller is responsible for sending the data, maybe it should also be responsible for managing the state, and this component just receives the values as props?

Reply:
If we want to keep both files while we are not sure that RconSettings provides a promise to Blacklist, then maybe it's better to assume the request succeeded and clear the fields regardless of the response from the server. Anything else would require RconSettings to micromanage the state of Blacklist.

Reply:
That's ok though? It's the spirit of High Order Component. where children component are 'dumb' ones doing render only.

Reply:
In my latest version I made it a fully controlled component, please check it again to see if I made any mistakes. It has some downsides because the rconSettings.js becomes feature envy, but it's best if we want to keep Blacklist.js as mostly render only.